code/frontend/src/components/MovieQueryFormComponent.js

```diff
@@ -23,13 +23,25 @@ function MovieQueryFormComponent({ setData, setLoading }) {
     const handleSubmit = async (e) => {
         e.preventDefault();  // Prevent default form submission behavior
 
-        // Get input values from the dynamically loaded form
-        const input1 = document.getElementById('input1').value;
-        const input2 = document.getElementById('input2').value;
+        const data_request = {
+            // Get input values from the dynamically loaded form
+            rank : document.getElementById('rank').value;
```

Comment:
you need to replace the ; -> ,

Reply:
you need to do this for all elements in the json. 
Look at an example json object. each element is separated by a comma not a semi-colon